.circleci/config.yml

```diff
@@ -30,64 +33,90 @@ commands:
           paths:
             - ~/android-tools
-
-  create-launch-android-emulator:
-    description: "create and launch android emulators"
-    steps:
-      - run:
-          name: create AVD
-          command: |
-            echo "no" | avdmanager --verbose create avd --force \
-              --name "Pixel_3a_API_26" \
-              --package "system-images;android-26;google_apis_playstore;x86"
-
-      - run:
-          name: start AVD
-          command: emulator @Pixel_3a_API_26 -no-window -no-audio
-          background: true
-
-      - run:
-          name: wait for emulator
-          command: |
-            adb wait-for-device shell 'while [[ -z $(getprop dev.bootcomplete) ]]; do sleep 1; done;'
   
   npm-dependencies:
-    description: "installs all global and local npm dependencies"
+    description: "Install all global and local npm dependencies"
     steps:
       - run: npm install -g cordova
       - run: npm install -g github:apache/cordova-paramedic
+      - run: npm install --legacy-peer-deps
+
+  npm-ios-dependencies:
+    description: "Install all ios global and local npm dependencies"
+    steps:
       - run: npm install -g ios-deploy
-      - restore_cache:
-          keys:
-            - v2-dependencies-{{ checksum "package.json" }}
-            - v2-dependencies-
-      - run: npm install
-      - save_cache:
-          paths:
-            - node_modules
-          key: v2-dependencies-{{ checksum "package.json" }}
+      - run: npm install --legacy-peer-deps
     
+  replace-api-key-linux:
+    description: "Replace API_KEY"
+    steps:
+      - run: sed -i s/api_key/$API_KEY/ tests/tests.js
+
   replace-api-key:
-    description: "replace API_KEY"
+    description: "Replace API_KEY"
     steps:
       - run: sed -i .bck s/api_key/$API_KEY/ tests/tests.js
 
+  install-sdkman:
+    description: Install SDKMAN
+    steps:
+      - run:
+          name: Installing SDKMAN
+          command: |
+            curl -s "https://get.sdkman.io?rcupdate=false" | bash
+            echo -e '\nsource "/home/circleci/.sdkman/bin/sdkman-init.sh"' >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: Setup Java environment
+          command: |
+            sdk env install
+            echo -e '\nexport JAVA_HOME="/home/circleci/.sdkman/candidates/java/current"' >> $BASH_ENV
+
+  install-nvm:
+    steps:
+      - run:
+          name: Install node 16.13.1
+          command: |
+            curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+            echo 'export NVM_DIR="$([ -z "${XDG_CONFIG_HOME-}" ] && printf %s "${HOME}/.nvm" || printf %s "${XDG_CONFIG_HOME}/nvm")"' >> $BASH_ENV
+            echo '[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"' >> $BASH_ENV
+      - run: nvm install 16.13.1
+      - run: nvm use 16.13.1
+      - run: echo 'nvm use 16.13.1' >> $BASH_ENV
+      - run: node -v
```

Comment:
This could also be moved into our own orb if we wanted to reuse in other projects